core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala

```diff
@@ -317,14 +321,19 @@ case class RangeEventDescriptorV0TLV(
       unitSize.bytes ++ unitBytes ++ precision.bytes
   }
 
-  lazy val outcomes: Vector[Int32] = {
+  override lazy val outcomes: Vector[String] = {
     val startL = start.toLong
     val stepL = step.toLong
 
     val range =
-      0L.until(count.toLong).map(num => startL + (num * stepL))
-
-    range.map(Int32(_)).toVector
+      0L.until(count.toLong)
+        .map { num =>
+          val double: Double =
+            (startL + (num * stepL)) * Math.pow(10, precision.toInt)
+          double.toString
+        }
+
+    range.map(i => s"$i").toVector
   }
```

Comment:
I think this should be split into 2 functions, one that is how it was so the oracle can just call `.outcomes` and get what it needs to sign and then add a second val called `.preciseOutcomes` that takes into account the precision for the actual values that are being represented.